From 482cb62cced016afefeb22797ec4b90298036873 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 25 Jul 2019 13:11:15 +0200
Subject: [PATCH 360/386] mmc: mmc_oops: increase record size and pick higher
 start-offset

Increasing record size results with capturing bigger context. The
chosen size is similar to one used for ramoops dump routine.
In addition instead of using low start-offset which may overwrite the
important data of the beginning of mmc (e.g. partition table)
chose 7GB offset.

Change-Id: I64adc1bf106345a439e29c887b083b6255d62a07
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12985
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 Documentation/devicetree/bindings/mmc/mmcoops.txt | 7 +++----
 drivers/mmc/core/mmc_oops.c                       | 2 +-
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/mmc/mmcoops.txt b/Documentation/devicetree/bindings/mmc/mmcoops.txt
index ba4373b49c39..a3f02e470a6d 100644
--- a/Documentation/devicetree/bindings/mmc/mmcoops.txt
+++ b/Documentation/devicetree/bindings/mmc/mmcoops.txt
@@ -6,8 +6,7 @@ Introduction and concept are similar to other oops logger.
 - Disadvantage: if MMC is occurred an oops/panic, this logger can't work.
 
 After reboot, you can get the last log with below command.
-#dd if=/dev/mmcblk0 of=/tmp/dump.log skip=1024 count=16
-#vi dump.log
+#dd if=/dev/mmcblk0 of=/tmp/dump.log skip=14680064 count=20; cat /tmp/dump.log
 
 Required Properties:
 
@@ -18,6 +17,6 @@ Required Properties:
 Example:
 	mmcoops {
 		compatible = "mmcoops";
-		start-offset = <64>;	/* 64 * 512B = 32KiB offset */
-		size = <16>;		/* 16 * 512B = 8KiB */
+		start-offset = <14680064>;	/* 14680064 * 512B = 7GB */
+		size = <20>;			/* 20 * 512B = 10KiB */
 	};
diff --git a/drivers/mmc/core/mmc_oops.c b/drivers/mmc/core/mmc_oops.c
index ceed30a61b84..4eab812d9d21 100644
--- a/drivers/mmc/core/mmc_oops.c
+++ b/drivers/mmc/core/mmc_oops.c
@@ -29,7 +29,7 @@
 #define MMCOOPS_KERNMSG_HDR	"===="
 #define MMCOOPS_HEADER_SIZE	(5 + sizeof(struct timeval))
 
-#define RECORD_SIZE		4096
+#define RECORD_SIZE		10240UL
 
 static int dump_oops = 1;
 module_param(dump_oops, int, 0600);
-- 
2.17.1

